src/basic/log.c

```diff
@@ -588,6 +592,27 @@ static int log_do_header(
         return 0;
 }
 
+static void log_do_context(struct iovec *iovec, size_t iovec_len, size_t *n) {
+        assert(iovec);
+        assert(n);
+
+        if (iovec_len <= 1)
+                return;
+
+        LIST_FOREACH(ll, c, _log_context) {
+                STRV_FOREACH(s, _log_context->fields) {
```

Comment:
I'm afraid this needs to be `c->fields` ;(

Maybe add a test that this logging actually works in TEST-04-JOURNAL ?

Reply:
Hmm, let me extend systemd-cat with support for adding extra fields and use this new functionality there immediately. Then I can use that in TEST-04-JOURNAL to test that this works.